### ISSUE TITLE
ReferenceError: fetch is not defined

### ISSUE THREAD

**Issue by zareth-san** [AnomalyInnovations/serverless-bundle#204]
tl;dr: `serverless-bundle`'s webpack output calls a function `fetch` which is never defined

**background**: I have created a new typescript serverless project using [serverless-typescript-starter](https://github.com/AnomalyInnovations/serverless-typescript-starter), which uses `serverless-bundle`. I've created a scheduled service which makes use of the @ethersproject/web dependency, which internally uses node's `https` standard library function [https.request](https://nodejs.dev/learn/making-http-requests-with-nodejs).

**Problem**: When I try to invoke my lambda function, it fails with the error `ReferenceError: fetch is not defined`. The stack trace indicates the error-causing line exists in `node_modules/@ethersproject/web/lib.esm/geturl.js`, which I assume is generated by `webpack`. Specifically the line is:

`const response = yield fetch(href, request);`

**Questions**: Why is `serverless-bundle`'s webpack outputing the call to `fetch`? I have read that `fetch` is a browser-specific function meant for making http requests, but then why is this used in a typescript-based lambda function? Most importantly, how do I instruct `serverless-bundle` to use the correct http request function calls, and not use `fetch`?

**Comment by zareth-san** [comment]
This is likely a problem with ethers.js, not serverless-bundle. So closing.

**Comment by yspreen** [comment]
I have the same problem.
